primitives/core/src/offchain/mod.rs

```diff
@@ -495,6 +497,9 @@ pub trait Externalities: Send {
 		buffer: &mut [u8],
 		deadline: Option<Timestamp>
 	) -> Result<usize, HttpError>;
+
+	/// Set the reserved nodes
```

Comment:
This needs way more documentation that just `Set the reserved nodes`.

- What is a reserved node?
- What happens in case of forks? (I actually don't know the answer to this one but I think it's important)
- What does "reserved only" mean?
- What is this `Vec<Vec<u8>>`? I think it's understandable that the outer one is a list, but the inner one is opaque and should be documented.

I'm also a bit wary of naming this "reserved nodes".
The problem with reserved nodes is that they're already configurable through the CLI and through the JSON-RPC endpoint.
Does it make sense for a user to configure reserved nodes through the CLI only for their settings to be erased by the chain immediately after? In my opinion, no.

As mentioned, we should not use reserved nodes themselves but another mechanism in parallel of reserved nodes. This requires changes in the peerset, which I'm fine with not doing now. However, if we name this "reserved nodes" we forever introduce an ambiguity in this concept, even if we later change the peerset.

Reply:
> What happens in case of forks? (I actually don't know the answer to this one but I think it's important)

After some questions and brainstorming, forks can probably not be mentioned.

If we assume that the runtime is written in such a way the list of reserved nodes is modified only through transactions, which is how I think this feature would be used, then they're not problematic.

The only thing that can differ between two forks is which transactions have been included and in which order.
Assuming that a transaction can allow access to a new `PeerId`, it is possible that a node receives a block that authorizes this new `PeerId`, then a reorg happens and this new `PeerId`s gets denied access again. But then sooner or later, the transaction that adds new the `PeerId` will be added to the chain.

Reply:
Of course, if the runtime does something stupid, such as randomly authorizing `PeerId`s based on the parent block hash or something like that, then forks can be problematic in the sense that we can give access to `PeerId`s that shouldn't be given access.

However I can't think of any reasonable scenario where that would be a problem.